packages/clay-link/src/index.tsx

```diff
@@ -85,14 +131,28 @@ const ClayLink = React.forwardRef<HTMLAnchorElement, IProps>(
 				[`btn-${displayType}`]: displayType && !outline && !borderless,
 				[`btn-outline-${displayType}`]:
 					displayType && (outline || borderless),
+				[weight ? FONT_WEIGHTS[weight] : '']: weight,
+				[`text-${fontSize}`]: fontSize,
+				[`text-decoration-${decoration}`]: decoration,
 			};
 		} else {
+			if (displayType && LINK_PRESETS[displayType]) {
+				if (decoration !== null && !outline) {
+					decoration =
+						decoration || LINK_PRESETS[displayType]?.decoration;
+				}
+			}
```

Comment:
I think we can make this a little simpler.

```suggestion
		decoration = decoration === null || outline ? undefined : decoration || LINK_PRESETS[displayType!]?.decoration;
```

We must rely on typing here so we don't have to do checks that wouldn't be expected because when the developer tries to compile the build will fail. Also using `?` and trying to access the object with an `undefined` value will have the same result as the above conditionals which are checking if the value exists in `LINKS_PRESETS` and if the `displayType` exists.